### PR TITLE
Change Availability of static actions to not accept CBs

### PR DESCRIPTION
**Pull request by ps37**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [X] Examples have been added / updated (for bug fixes / features)
-   [X] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [X] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [ ] Other... Please describe:

## What does this change do?
Callbacks with selected entities are only meant for contextual actions and users were using them for static actions also. So now with this change, when a user tries to assign values to the availability of static actions other than observables, they get an error.

## What manual testing did you do?
- Fixed the components of examples project which are wrongly using the availability callbacks
- Started the examples app and made sure that the examples render as before
- Copy pasted the code to node modules of vcd ui and noticed that the code in fact shows an error when a callback is assigned to availability
of static actions

## Screenshots (if applicable)
Screen Shot 2021-04-07 at 8 14 37 AM
Screen Shot 2021-04-07 at 8 13 20 AM

## Does this PR introduce a breaking change?

-   [X] Yes
-   [ ] No

Any existing static actions in VCD ui that use callbacks for their availability are now going to break. As a follow up to this task, I am going to fix all those places in the VCD ui

### CODE REVIEW

File: projects/components/CHANGELOG.MD [hunk 1]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 - Added functionality to disable an option in `FormSelectComponent`
 
+## [2.0.0-dev.8]
+### Changed
+- Changed the interface action availability such that the availability of static actions does not accept call backs
```
Comment (ryan-bradford) [inline projects/components/CHANGELOG.MD:13]: Let's label this with breaking

File: projects/components/src/action-menu/action-menu.component.ts [hunk 2]
```diff
@@ -23,6 +30,21 @@ export function getDefaultActionDisplayConfig(cfg: ActionDisplayConfig = {}): Ac
...
+ */
+interface ActionItemInternal<R, T> extends BaseActionItem<R, T> {
+    /**
+     * Used for determining where in the action menu this action gets displayed
+     */
+    actionType?: ActionType;
+    /**
+     * Condition whether or not the action is available.
+     */
+    availability?: ((selectedEntities: R[]) => boolean) | Observable<boolean> | boolean;
```
Comment (ryan-bradford) [inline projects/components/src/action-menu/action-menu.component.ts:45]: I thought we decided that `((selectedEntities: R[]) => boolean)` shouldn't be in the internal availability?

To get around the children problem, you might want to try redeclaring `children` in this interface with the type `ActionItemInternal`.

File: projects/examples/src/app/components/action-menu/with-separators/action-menu-with-separators.example.component.html [hunk 3]
```diff
@@ -12,6 +12,7 @@
         [actions]="actions"
         [actionDisplayConfig]="actionDisplayConfig"
         [dropdownTriggerBtnText]="'vcd.cc.action.menu.actions'"
+        [selectedEntities]="[{}]"
```
Comment (ryan-bradford) [inline projects/examples/src/app/components/action-menu/with-separators/action-menu-with-separators.example.component.html:15]: Why did you need to add this?

**Review by ryan-bradford — CHANGES_REQUESTED** [review]
Can you add screenshots from vcd_ui showing the actions still working?

File: projects/components/src/action-menu/action-menu.component.ts (hunk 2 above)
Comment (ps37) [reply projects/components/src/action-menu/action-menu.component.ts:45]: >To get around the children problem, you might want to try redeclaring children in this interface with the type ActionItemInternal.
That is not working as TS complains that action types are incompatible

File: projects/examples/src/app/components/action-menu/with-separators/action-menu-with-separators.example.component.html (hunk 3 above)
Comment (ps37) [reply projects/examples/src/app/components/action-menu/with-separators/action-menu-with-separators.example.component.html:15]: Without selected entities, contextual actions will not be shown and in this example, we need the contextual actions to show the menu with separators.

File: projects/components/src/action-menu/action-menu.component.ts (hunk 2 above)
Comment (ryan-bradford) [reply projects/components/src/action-menu/action-menu.component.ts:45]: I still wanna push back on this. We shouldn't be modifying the definition of `availability` to get some Typescript issue to go away. Having the callback still in this type is confusing and could lead to problems down the line. 

Maybe children need to be given a type of `BaseActionItem`? And then in here they're redeclared to be `ActionItemInternal`? Like, there's some way to rework these types so they pass without any casts or hacks.

Comment (ps37) [reply projects/components/src/action-menu/action-menu.component.ts:45]: I removed the call back from ActionItemInternal. However, I still required the casts in 2 places. I think that is fine because, at those points in the code, the action children are converted into ActionItemInternal but TS doesn't know that and we don't want to expose children as ActionItemInternal.

File: projects/components/CHANGELOG.MD [hunk 4]
```diff
@@ -8,7 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ## [2.0.0-dev.9]
-### Changed
+### Breaking
```
Comment (ryan-bradford) [inline projects/components/CHANGELOG.MD:11]: The format we generally use is 
```
### Changed
- BREAKING: Changed the interface action availability such that the availability of static actions does not accept call backs
```

File: projects/components/src/action-menu/action-menu.component.ts [hunk 5]
```diff
@@ -34,15 +34,15 @@ export function getDefaultActionDisplayConfig(cfg: ActionDisplayConfig = {}): Ac
  * We internally convert the callbacks to booleans to avoid calling the callbacks all the time from template. However, we don't want to
  * allow callers to assign boolean variables to availability as there is no way to know when those variables can get updated from outside.
  */
-interface ActionItemInternal<R, T> extends BaseActionItem<R, T> {
+export interface ActionItemInternal<R, T> extends BaseActionItem<R, T> {
```
Comment (ryan-bradford) [inline projects/components/src/action-menu/action-menu.component.ts:37]: Why export this?

**Review by ryan-bradford — APPROVED** [review]
Small comments + add screenshots from VCD_UI
